docs/docs/life-and-times-of-a-gatsby-build.md

```diff
@@ -2,7 +2,311 @@
+
+So what's the difference?
+
+If you compare the outputs of the two commands (`gatsby develop` vs `gatsby build`), you can see that everything (with the exception of deleting [HTML](/docs/glossary#html) and [CSS](/docs/glossary#css) files) up until the line that says `info bootstrap finished` are the same. However, `gatsby build` runs some additional steps to prepare your site to go live after the bootstrap phase.
+
+The following output shows the differences between the above two examples:
+
+```diff
+success open and validate gatsby-configs - 0.051 s
+success load plugins - 0.915 s
+success onPreInit - 0.021 s
++ success delete html and css files from previous builds - 0.030 s
+success initialize cache - 0.034 s
+success copy gatsby files - 0.099 s
+success onPreBootstrap - 0.034 s
+success source and transform nodes - 0.121 s
+success Add explicit types - 0.025 s
+success Add inferred types - 0.144 s
+success Processing types - 0.110 s
+success building schema - 0.365 s
+success createPages - 0.016 s
+success createPagesStatefully - 0.079 s
+success onPreExtractQueries - 0.025 s
+success update schema - 0.041 s
+success extract queries from components - 0.333 s
+success write out requires - 0.020 s
+success write out redirect data - 0.019 s
+success Build manifest and related icons - 0.141 s
+success onPostBootstrap - 0.130 s
+⠀
+info bootstrap finished - 3.674 s
+⠀
+success run static queries - 0.057 s — 3/3 89.08 queries/second
+success run page queries - 0.033 s — 5/5 347.81 queries/second
+- success start webpack server - 1.707 s — 1/1 6.06 pages/second
++ success Generating image thumbnails — 6/6 - 1.059 s
++ success Building production JavaScript and CSS bundles - 8.050 s
++ success Rewriting compilation hashes - 0.021 s
++ success Building static HTML for pages - 0.852 s — 4/4 23.89 pages/second
++ info Done building in 16.143999152 sec
+```
+
+_**Note**: the output of `gatsby develop` and `gatsby build` can vary based on plugins you've installed that can tap into the build lifecycle and the [Gatsby reporter](/docs/node-api-helpers/#reporter). The above output is from running the default Gatsby starter._
+
+There is only one difference in the bootstrap phase where HTML and CSS is deleted to prevent problems with previous builds. In the build phase, the build command skips setting up a dev server and goes into compiling the assets.
+
+By omitting these later steps, `gatsby develop` can speed up your ability to make live edits without page reloads using features like [hot module replacement](/docs/glossary#hot-module-replacement). It also saves time with the more CPU intensive processes that aren't necessary to perform in rapid development.
+
+A [cache](/docs/glossary#cache) is also used to detect changes to `gatsby-*.js` files (like `gatsby-node.js`, or `gatsby-config.js`) or dependencies. It can be cleared manually with the `gatsby clean` command to work through issues caused by outdated references and a stale cache.
```

Comment:
```suggestion
A [cache](/docs/glossary#cache) is also used to detect changes to `gatsby-*.js` files (like `gatsby-node.js`, `gatsby-config.js`), or dependencies. It can be cleared manually with the `gatsby clean` command to work through issues caused by outdated references and a stale cache.
```